src/apps/js_generic/js_generic.cpp

```diff
@@ -365,9 +372,23 @@ namespace ccfapp
         JS_SetPropertyStr(ctx, global_obj, "query", query_str);
 
         const auto& request_body = args.rpc_ctx->get_request_body();
-        auto body_str = JS_NewStringLen(
-          ctx, (const char*)request_body.data(), request_body.size());
-        JS_SetPropertyStr(ctx, global_obj, "body", body_str);
+        auto content_type = args.rpc_ctx->get_request_header("content-type");
```

Comment:
Let's talk about this, but I think we'd like to pass the ArrayBuffer every time and let the JS code decide how it wants to deal with wrt content-type, rather than manage it on the C++ side, unless there's a good reason to do it otherwise.

Reply:
I'll try to make body something like a [`Body`](https://developer.mozilla.org/en-US/docs/Web/API/Body).

Reply:
Done, can you have another look?

Reply:
The new scheme looks excellent :)